ui/management/commands/retranscode_collections.py

```diff
@@ -0,0 +1,271 @@
+
+        # Apply date filters if specified
+        base_queryset = self._apply_date_filters(base_queryset, options)
+
+        if options["ids"]:
+            ids = options["ids"]
+            collections_by_pk = base_queryset.filter(
+                pk__in=[int(id) for id in ids if id.isdigit()]
+            )
+            return collections_by_pk.distinct()
+
+        elif options["all"]:
+            return base_queryset
+
+        elif options["owner"]:
+            try:
+                user = User.objects.get(username=options["owner"])
+                return base_queryset.filter(owner=user)
+            except User.DoesNotExist:
+                raise CommandError(f"User '{options['owner']}' does not exist")
+
+        elif options["course_ids"]:
+            return base_queryset.filter(edx_course_id__in=options["course_ids"])
+
+        elif options["edx_endpoint"]:
+            return base_queryset.filter(
+                edx_endpoints__name=options["edx_endpoint"]
+            ).distinct()
+
+        return base_queryset.prefetch_related("edx_endpoints", "videos")
+
+    def _apply_date_filters(self, queryset, options):
+        """Apply created_at date filters to the queryset"""
+        if options.get("created_after"):
+            created_after = self._parse_date(options["created_after"], "created-after")
+            queryset = queryset.filter(created_at__gte=created_after)
+
+        if options.get("created_before"):
+            created_before = self._parse_date(
+                options["created_before"], "created-before"
+            )
+            queryset = queryset.filter(created_at__lte=created_before)
+
+        return queryset
+
+    def _parse_date(self, date_string, argument_name):
+        """Parse date string and return timezone-aware datetime"""
+        try:
+            # Try parsing as full datetime first
+            parsed_date = parse_datetime(date_string)
+            if parsed_date:
+                # If no timezone info, assume UTC
+                if timezone.is_naive(parsed_date):
+                    parsed_date = timezone.make_aware(parsed_date, timezone.utc)
+                return parsed_date
+
+            # Try parsing as date only (YYYY-MM-DD)
+            try:
+                parsed_date = datetime.strptime(date_string, "%Y-%m-%d")
+                # For "created-after", use start of day; for "created-before", use end of day
+                if argument_name == "created-before":
+                    parsed_date = parsed_date.replace(
+                        hour=23, minute=59, second=59, microsecond=999999
+                    )
+                return timezone.make_aware(parsed_date, timezone.utc)
+            except ValueError:
+                pass
+
+            raise CommandError(
+                f"Invalid date format for --{argument_name}: '{date_string}'. "
+                f"Use YYYY-MM-DD or YYYY-MM-DD HH:MM:SS format."
+            )
```

Comment:
[nitpick] Why pass and then immediately raise a command error?

````suggestion
            except ValueError:
                raise CommandError(
                    f"Invalid date format for --{argument_name}: '{date_string}'. "
                    f"Use YYYY-MM-DD or YYYY-MM-DD HH:MM:SS format."
            )
````